app/src/main/java/com/github/sdp_begreen/begreen/fragments/UserPhotosViewAdapter.kt

```diff
@@ -44,41 +58,43 @@ class UserPhotosViewAdapter(
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
 
-        val photo = photos?.get(position)
+        lifecycleScope.launch {
 
-        if(isFeed) {
+            val photo = photos?.get(position)
+            val userId = photo?.takenBy ?: return@launch
 
-            // Display avatar if on feed
-            val drawable = ContextCompat.getDrawable(holder.avatarView.context, R.drawable.ic_baseline_person)
-            val defaultAvatar = drawable?.toBitmap()
-            // holder.avatarView.setImageBitmap(getFromDB(photo) ?: defaultAvatar)
-            holder.avatarView.setImageBitmap( defaultAvatar)
+            val user = db.getUser(userId)
 
-        }else{
-            // Do not display avatar if not on feed
-            holder.avatarView.visibility = View.GONE
-        }
+            if (isFeed) {
 
-        if (photo is TrashPhotoMetadata) {
-            // use the photo variable as a TrashPhotoMetadata instance
-
-            //Set default value
-            holder.titleView.text = photo?.caption ?: "No title"
-            holder.subtitleView.text = (photo?.takenOn?.toString()
-                ?: "Unknown date") + " | " + (photo?.trashCategory?.title
-                ?: "No category")
-            //holder.photoView.setImageBitmap(photo.getPhotoFromDataBase())
-            //TODO------------FOR DEMO -----------------
-            val url = URL("https://picsum.photos/400")
-            holder.photoView.setImageBitmap(
-                BitmapFactory.decodeStream(
-                    url.openConnection().getInputStream()
-                )
-            )
-            //------------FOR DEMO -----------------
+                holder.avatarView.visibility = View.VISIBLE
+
+                // Display avatar if on feed
+                if (user?.profilePictureMetadata != null) {
```

Comment:
Here I would recommend you to write it like this. It prevents you from using the `!!` to assert a non-null value on the profilePictureMetadata.

```suggestion
                user?.profilePictureMetadata?.also {
                    val avatarImage = db.getUserProfilePicture(it, userId)
                    holder.avatarView.setImageBitmap(avatarImage)
                }
               
```

Reply:
You are right, it cleaner ! Thank you